paddle/fluid/framework/parallel_executor.cc

```diff
@@ -258,81 +264,118 @@ ParallelExecutor::ParallelExecutor(
 
   // Step 2. Convert main_program to SSA form and dependency graph. Also, insert
   // ncclOp
-  std::unique_ptr<ir::Graph> graph;
+  std::vector<std::unique_ptr<ir::Graph>> graphs;
 #if defined(PADDLE_WITH_CUDA) && !defined(_WIN32)
-  graph = build_strategy.Apply(main_program, member_->places_, loss_var_name,
-                               member_->local_scopes_, member_->nranks_,
-                               member_->use_cuda_, member_->nccl_ctxs_.get());
+  if (build_strategy.async_mode_ && !build_strategy.is_distribution_) {
+    VLOG(3) << "use local async mode";
+    for (size_t i = 0; i < member_->places_.size(); ++i) {
```

Comment:
@panyx0718 has a PR that passed graph insetead of program: https://github.com/PaddlePaddle/Paddle/pull/15425 . And ParallelGraphExecutor does not dependence program_desc: https://github.com/PaddlePaddle/Paddle/pull/15716 ?